test/jdk/java/net/httpclient/DigestEchoClient.java

```diff
@@ -508,7 +526,10 @@ void testBasic(Version clientVersion, Version serverVersion, boolean async,
                 if (queue.remove(100) == ref) break;
             }
             var error = TRACKER.checkShutdown(900);
-            if (error != null) throw error;
+            if (error != null) {
+                if (failed != null) error.addSuppressed(failed);
+                throw error;
```

Comment:
Do you think an additional `error != failed` check would be needed here, before calling `addSuppressed()` or is that not a concern here?

Reply:
`error` is a brand new AsssertionError created by the tracker. It simply can't be the same exception as `failed`.

Reply:
You are right - I missed that `error` is declared and assigned just one line above this and that can only be an `AssertionError`.